### PR TITLE
chore: adding distribute fee events to ics29

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #2847 

### Commit Message / Changelog Entry

```bash
chore: adding distribute fee events to ics29
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by chatton — APPROVED** [review]
Nice and clean LGTM 👍

**Review by crodriguezvega — COMMENTED** [review]
Should we extend the existing tests to use the brand new [`AssertEvents` function](https://github.com/cosmos/ibc-go/blob/f24f41ea8a61fe87f6becab94e84de08c8aa9381/testing/events.go#L138) to check that the events emitted are what we expect?

**Comment by damiannolan** [comment]
> Should we extend the existing tests to use the brand new [`AssertEvents` function](https://github.com/cosmos/ibc-go/blob/f24f41ea8a61fe87f6becab94e84de08c8aa9381/testing/events.go#L138) to check that the events emitted are what we expect?

Most of the existing tests call into the keeper functions directly. To get at the events we'll need to route a new `MsgAcknowledgement` with the testing library's `chain.SendMsgs(...)` method. I can look at adding a test to events_test.go but it might take a good bit of boilerplate to generate a proof etc..

File: modules/apps/29-fee/keeper/events_test.go [hunk 1]
```diff
@@ -81,3 +86,80 @@ func (suite *KeeperTestSuite) TestIncentivizePacketEvent() {
 		}
 	}
 }
+
+func (suite *KeeperTestSuite) TestDistributeFeeEvent() {
```
Comment (crodriguezvega) [inline modules/apps/29-fee/keeper/events_test.go:90]: ❤️ Thanks, @damiannolan!

**Review by charleenfei — APPROVED** [review]
nice one, lgtm!

**Pull request merged**